File: src/serverless/BlazingPizza.Functions.CSharp/GetOrders.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;
using System.Linq;
using System.Security.Claims;

namespace BlazingPizza.Functions.CSharp
{
    public static class GetOrders
    {
        [FunctionName("get-orders")]
        public static IActionResult Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "orders")] HttpRequest req,
            [CosmosDB("blazingPizza",
                      "pizza",
                      ConnectionStringSetting = "CosmosConnectionString",
                      SqlQuery = "SELECT * FROM c WHERE IS_DEFINED(c.OrderId)")] IEnumerable<Order> orders,
            ILogger log)
        {
            var principal = StaticWebAppsAuth.Parse(req);
            var userId = principal.Claims.First(c => c.Type == ClaimTypes.NameIdentifier);

            var ordersForUser = orders.Where(o => o.UserId == userId.Value).Select(OrderWithStatus.FromOrder);

            return new OkObjectResult(ordersForUser);
        }
    }
}

File: src/serverless/BlazingPizza.Functions.FSharp/GetOrders.fs
namespace BlazingPizza.Functions.FSharp

open Microsoft.AspNetCore.Mvc
open Microsoft.Azure.WebJobs
open Microsoft.Azure.WebJobs.Extensions.Http
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.Logging
open System.Security.Claims

open BlazingPizza

module GetOrders =
    [<FunctionName("get-orders")>]
    let runGetOrders ([<HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "orders")>] req: HttpRequest)
                     ([<CosmosDB("blazingPizza",
                                 "pizza",
                                 ConnectionStringSetting = "CosmosConnectionString",
                                 SqlQuery = "SELECT * FROM c WHERE IS_DEFINED(c.OrderId)")>] placedOrders: Order seq)
                     (log: ILogger)
                     =
        let principal = ClientPrincipal.parse req

        let userId = principal.Claims |> Seq.find(fun c -> c.Type = ClaimTypes.NameIdentifier)

        let userOrders =
            placedOrders
            |> Seq.filter (fun o -> o.UserId = userId.Value)
            |> Seq.map OrderWithStatus.FromOrder

        OkObjectResult userOrders
